core/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -1020,6 +1025,19 @@ public synchronized void updateRawTableMetadata(int id, ByteBuffer metadata) thr
    * @param uri The uri to validate
    */
   private void validateUri(TachyonURI uri) throws IOException {
+    if (!uri.isPathAbsolute() && !TachyonURI.EMPTY_URI.equals(uri)) {
+      LOG.debug("URL is invalid because of Absolute");
+    }
+    if (uri.hasScheme() && !mRootUri.getScheme().equals(uri.getScheme())) {
+      LOG.debug("URL is invalid because of schema");
+    }
+
+    if (uri.hasAuthority() && !mRootUri.getAuthority().equals(uri.getAuthority())) {
+      LOG.debug("URL is invalid because of authority. URI has authoring "
+              + uri.hasAuthority() + " root authoring " + mRootUri.getAuthority()
+              + " uri authority " + uri.getAuthority());
+    }
+
```

Comment:
If the logic is done here, the following if statement can be simplified. `uri == null` should be checked first though.